docs/kb/semgrep-code/scan-engine-kill.md

```diff
@@ -0,0 +1,146 @@
+  <multilang>      60   199251          Community     551
+  ts              166    26672          Pro rules     498
+  python          314     8089          Custom         19
+  scala            13     5415
+  json              1     4149       
+  yaml              7     1952
+  js              160     1084      
+  terraform        13      470      
+  bash              4      408           
+  go               95      228
+  ruby             22      228
+  php              25       99
+  swift            41       89
+  html              1       72         
+  dockerfile        2       31           
+  c                 9       16
+  rust             49        8
+  java            156        2
+  kotlin           47        1
+```
+
+Now you have a good idea of the size of your monorepository. After establishing the size and breakdown of programming languages and files, you can decide what adjustments to take for a scan to succeed.
+
+## Scanning components separately 
+
+Based on the composition provided by the logs, you may be able to determine if your repository is modular. If so, you can try scanning the components separately.
+
+Interfile analysis still applies in the smaller scope assuming the modules are disparate. 
+
+### Scanning schedule
+
+You can perform modular scans on a daily basis and then perform one big monolithic scan every two weeks or so to keep close tabs and cap any false negative interfile or inter-component outliers that can get introduced with code change as a way of truing up your dependency graph for the monolithic build. This helps deflect costs as you are scanning the majority on smaller runners/k8s.  
+
+However, you still need to scan the whole monolithic build, whether you scan it biweekly or daily, and the following sections apply to tackling that monolith whether you adopt the modular paradigm of scanning or not. 
+
+## Serializing types of scans
+
+With Semgrep 1.42.0 and later, Secret rulesets are now included in the default scan configuration.  Instead of scanning Code, Supply Chain and Secret rulesets concurrently, you can circumnavigate resource limits by running different types of scans in a serial fashion.  Rather than issuing one monolithic command, for example:
+
+```console
+ $ semgrep ci
+```
+ 
+You can instead run:
+ 
+ ```
+$ semgrep ci --code --pro
+$ semgrep ci --supply-chain --oss-only
+$ semgrep ci --secrets
+ 
+ ```
+
+As a result, less memory is used in total at any point in time.
+
+## Increasing RAM 
+
+Lastly, you can also tackle a large scan by increasing the RAM.
+
+### Establish RAM baseline and avoid swap memory
+
+First, establish how much memory is required to scan. Determining the total amount of memory required is key, not just to avoid killed scans but running scans that use swap memory. Avoid swap memory when scanning with any static analysis tool. Semgrep and other SAST tools are disk I/O intensive and swapping in and out with a swap file reduces performance severely.
```

Comment:
```suggestion
First, establish how much memory is required to scan. Determining the total amount of memory required is key, not just to avoid killed scans but to run scans that use swap memory. Avoid swap memory when scanning with any static analysis tool. Semgrep and other SAST tools are disk I/O intensive and swapping in and out with a swap file reduces performance severely.
```